pkg/util/pki/csr.go

```diff
@@ -319,11 +319,36 @@ func GenerateCSR(crt *v1.Certificate, optFuncs ...GenerateCSROption) (*x509.Cert
 	}
 
 	if opts.EncodeNameConstraintsInRequest && crt.Spec.NameConstraints != nil {
-		extension, err := MarshalNameConstraints(crt.Spec.NameConstraints)
-		if err != nil {
-			return nil, err
+		nameConstraints := &NameConstraints{}
+
+		if crt.Spec.NameConstraints.Permitted != nil {
+			nameConstraints.PermittedDNSDomains = crt.Spec.NameConstraints.Permitted.DNSDomains
+			nameConstraints.PermittedIPRanges, err = parseCIDRs(crt.Spec.NameConstraints.Permitted.IPRanges)
+			if err != nil {
+				return nil, err
+			}
+			nameConstraints.PermittedEmailAddresses = crt.Spec.NameConstraints.Permitted.EmailAddresses
+			nameConstraints.ExcludedURIDomains = crt.Spec.NameConstraints.Permitted.URIDomains
```

Comment:
Shouldn't this be `nameConstraints.PermittedURIDomains` ? @tanujd11 @inteon

Reply:
Indeed, good catch!
If this bug is still present on master, could you maybe create a PR to fix this?

Reply:
It is, will create a PR

Reply:
Thanks @martin31821